lib/class/vainfo.class.php

```diff
@@ -449,10 +450,12 @@ public static function clean_tag_info($results, $keys, $filename = null)
 
             $info['lyrics']    = $info['lyrics']
                     ?: strip_tags(nl2br((string) $tags['lyrics']), "<br>");
-            $info['replaygain_track_gain'] = $info['replaygain_track_gain'] ?: floatval($tags['replaygain_track_gain']);
-            $info['replaygain_track_peak'] = $info['replaygain_track_peak'] ?: floatval($tags['replaygain_track_peak']);
-            $info['replaygain_album_gain'] = $info['replaygain_album_gain'] ?: floatval($tags['replaygain_album_gain']);
-            $info['replaygain_album_peak'] = $info['replaygain_album_peak'] ?: floatval($tags['replaygain_album_peak']);
+            $info['replaygain_track_gain'] = $info['replaygain_track_gain'] ?: $tags['replaygain_track_gain'];
```

Comment:
it's possibly in the ternary's here.

null and 0 will count as false here and the ?: means if it's not true it will be null so it's probably skipping 0's here

Reply:
To be honest I don't understand that clean_tag_info function...

If its initialising $info as an empty array, how can the ternary evaluate to anything other than false? Argh! lol

With that said I've already tried wrapping them in ```!is_null```, ```!empty```, ```isset``` and they all seem to result in the values coming back null (or 1...)

Here is my log I kept as I delved into the depths printing out before and after debug lines trying to pinpoint where the 0 is turning to null...

```
local.catalog.php insert_local_song()

seems to be happening in $vainfo->get_info(); in local.catalog.php -> insert_local_song()

back in vainfo... _get_tags

_parse_general the culprit?

clean_tag_info...?

definitely in $vainfo->get_info(); somewhere!

DEFINITELY 100% in _get_tags
```

Reply:
which array has the 0? infor or tags?
$info['replaygain_track_gain']
$tags['replaygain_track_gain']

Reply:
Neither does, that's the problem..

Let me set up a demonstration somehow